### PR TITLE
[PLAT-432] Allow admin app to poke stuck registrations

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

It would be more convenient to have admins be able to identify and restart registrations using a nice UI, instead of getting Steve to run a script. This fix adds that UI.

## Changes

- Adds list view with list of stuck registrations with UI changes so its linked on the front page.
- Adds view for restarting regs via a button and modal
- Adds tests for both
- Adds a not strictly necessary line of code which add localhost to allowed hosts when the app is running locally.

## QA Notes

This is a tricky one to test because you have to fake a stuck registration and a lot of the staging data doesn't behave realistically and will resist restarting. The process I used to stick the registrations locally was just to kill the worker container, create the registration and start the worker then restart the registration via the Admin app.

 If you're doing this on staging you should be to stick a registration by registering a huge github project, however their is currently a fix in the works for this so that might not work. Registering Googledrive projects with the files that have the same name/path also may cause them to become stuck.

A stuck registration must be stuck for one day before it appears on the stuck registration list, so we may want to have someone go into to staging and mock the dates so we don't have to wait to test it.

Also using old staging data dumps seems you produce more stuck registrations then could be realistically possible. Keep an eye out for that.

## Documentation

This is an admin only feature and I don't think we have any admin only docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-432

### CODE REVIEW

File: admin/templates/nodes/node.html [hunk 1]
```diff
@@ -15,93 +15,92 @@
...
+                                    {% csrf_token %}
+                                    <input class="btn btn-success" type="submit"
+                                           value="Restore Node" />
+                                </form>
+                            {% endif %}
+                        {% else %}
+                            <a href="{% url 'nodes:restart-stuck-registrations' guid=node.id %}"
+                               data-toggle="modal" data-target="#confirmRestartRegistration"
+                               class="btn btn-warning">
+                                Restart Registration
```
Comment (pattisdr) [inline admin/templates/nodes/node.html]: It looks like we give the ability to Restart Registration for every single registration.  The force_archive script does verifications that the force_archive can occur, so maybe this shouldn't be an option for all.

File: admin/templates/nodes/restart_registrations_modal.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<form class="well" method="post" action="{% url 'nodes:restart-stuck-registrations' guid=guid %}">
+    <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal">x</button>
+        <h3>Are you sure you want to restart the registration process? {{ guid }}</h3>
+    </div>
+    {% csrf_token %}
+    <div class="modal-footer">
+        <input class="btn btn-danger" type="submit" value="Confirm" />
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply admin/templates/nodes/restart_registrations_modal.html]: That's for future reference that is now addressed by the django messages package.

File: osf/models/registrations.py [hunk 3]
```diff
@@ -65,6 +67,17 @@ class Registration(AbstractNode):
                                                     null=True, blank=True,
                                                     on_delete=models.SET_NULL)
 
+    @staticmethod
+    def find_failed_registrations():
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/models/registrations.py]: 👍 done

File: admin/nodes/serializers.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ def serialize_node(node):
         'parent': node.parent_id,
         'root': node.root._id,
         'is_registration': node.is_registration,
+        'is_stuck_registration': getattr(node, 'is_stuck_registration', False),
```
Comment (pattisdr) [inline admin/nodes/serializers.py]: Where are you using this?

Comment (Johnetordoff) [reply admin/nodes/serializers.py]: [here] (https://github.com/CenterForOpenScience/osf.io/pull/8271/files#diff-5aee9e54ada2eec32fe347e858285b03R50) easy to miss.

**Comment by Johnetordoff** [comment]
@sloria  rebranched and conflicts dealt with.

**Pull request merged**